.circleci/config.yml

```diff
@@ -0,0 +1,79 @@
+# Use the latest 2.1 version of CircleCI pipeline process engine.
+# See: https://circleci.com/docs/2.0/configuration-reference
+version: 2.1
+
+# Define a job to be invoked later in a workflow.
+# See: https://circleci.com/docs/2.0/configuration-reference/#jobs
+jobs:
+  enforce-commit-standards:
+    docker:
+      - image: cimg/base:2022.09
+    steps:
+      - add_ssh_keys:
+          fingerprints:
+            - "55:47:0b:af:ad:97:89:06:19:4e:5c:34:06:50:f8:16"
+      - checkout
+      - run:
+          name: "Install Python and Java"
+          command: |
+            sudo apt update
+            sudo apt install software-properties-common
+            sudo add-apt-repository -y ppa:deadsnakes/ppa
+            sudo apt install python3.9 python3-pip openjdk-8-jdk
+      - run:
+          name: "Print Python and Java versions"
+          command: |
+            python3 -V
+            pip3 --version
+            java -version
+      - run:
+          name: Install pre-commit-checks
+          command: pip3 install pre-commit==2.16.0
+      - run:
+          name: Run pre-commit on all files
+          command: pre-commit run --all-files || true
+      - run:
+          name: Commit changes back to GitHub
+          command: |
+            if git status; then
+              git config --global user.email "rpanman@sonatype.com"
+              git config --global user.name "CircleCI"
+              git add *
+              git commit -m "Updated after pre-commit checks by CircleCI [skip ci]"
+              git push
+            fi
+
+  test-with-gradle:
+    # Specify the execution environment. You can specify an image from Dockerhub or use one of our Convenience Images from CircleCI's Developer Hub.
+    # See: https://circleci.com/docs/2.0/configuration-reference/#docker-machine-macos-windows-executor
+    docker:
+      - image: cimg/openjdk:11.0
+    # Add steps to the job
+    # See: https://circleci.com/docs/2.0/configuration-reference/#steps
+    steps:
+      - checkout
+      - run:
+          name: "Run tests"
+          command: |
+            java --version
+            ./gradlew codeCoverageReport
+      - save_cache:
+          paths:
+            - ~/.gradle/wrapper
+          key: v1-gradle-wrapper-{{ checksum "gradle/wrapper/gradle-wrapper.properties" }}
+      - save_cache:
+          paths:
+            - ~/.gradle/caches
+          key: v1-gradle-cache-{{ checksum "build.gradle" }}
+      - store_artifacts: # Upload test results for display in Artifacts: https://circleci.com/docs/artifacts/
+          path: build/reports/jacoco/codeCoverageReport/html
+
+# Invoke jobs via workflows
+# See: https://circleci.com/docs/2.0/configuration-reference/#workflows
+workflows:
+  develop-branch-workflow:
+    when:
```

Comment:
FWIW, one of my favorite features of CircleCI is the ability to run a CI build locally. It appear using `when` to determine which branch to build causes issues when trying to run a local build. Changing this to use `filters: branches: only` plays nice with local builds. Here's a snippet if you're interested:

```
workflows:
  develop-branch-workflow:
#    when:
#      equal: ['develop', << pipeline.git.branch >>]
    jobs:
      - enforce-commit-standards:
          filters:
            branches:
              only:
                - develop
```

With the above change, you could run a local build (after installing the CircleCI CLI tool - see: https://github.com/sonatype-nexus-community/nancy/blob/4a884909184b2888df89c2f596657b4cc88fb3de/.circleci/circleci-readme.md, and feel free to copy relevant parts of that `.circleci/circleci-readme.md` into this project if you like), you could run a local build using these commands:

```
circleci config process .circleci/config.yml > .circleci/local-config.yml
circleci local execute -c .circleci/local-config.yml --job 'enforce-commit-standards'
```

Given the above, you might also consider adding the `.circleci/local-config.yml` to the project `.gitignore` file:
```
# ci config for local ci build
.circleci/local-config.yml
```

BTW, the timing of this change happens to land when there is s known issue with local CircleCI builds and Docker  `vgroup2`, so please ping me if you try a local build and the build does not launch.

Reply:
Great, thanks Dan, I'll take a look - that looks like it would make writing new steps faster. In general I'm finding CircleCI much more pleasant to use than GitHub Actions or Jenkins. That might be because my first foray into CI/CD used Gitlab CI/CD and the YAML structure is similar, or it might just be the pretty GUI and good documentation :-)